src/faction_camp.cpp

```diff
@@ -2842,96 +2842,101 @@ void basecamp::start_fortifications( const mission_id &miss_id, float exertion_l
               "constructions." ) );
     tripoint_abs_omt start = om_target_tile( omt_pos, 2, 90, allowed_locations,
                              ot_match_type::type, true, omt_pos, _( "Select a start point from %d to %d tiles away." ) );
+    if( start == overmap::invalid_tripoint ) {
+        return;
+    }
     tripoint_abs_omt stop = om_target_tile( omt_pos, 2, 90, allowed_locations,
                                             ot_match_type::type,
                                             true, start, _( "Select an end point from %d to %d tiles away." ) );
-    if( start != tripoint_abs_omt( -999, -999, -999 ) &&
-        stop != tripoint_abs_omt( -999, -999, -999 ) ) {
-        const recipe &making = recipe_id( miss_id.parameters ).obj();
-        bool change_x = start.x() != stop.x();
-        bool change_y = start.y() != stop.y();
-        if( change_x && change_y ) {
-            popup( _( "Construction line must be straight!" ) );
-            return;
-        }
-        if( miss_id.parameters == faction_wall_level_n_1_string ) {
-            std::vector<tripoint_abs_omt> tmp_line = line_to( stop, start );
-            int line_count = tmp_line.size();
-            int yes_count = 0;
-            for( tripoint_abs_omt &elem : tmp_line ) {
-                if( std::find( fortifications.begin(), fortifications.end(), elem ) != fortifications.end() ) {
-                    yes_count += 1;
-                }
-            }
-            if( yes_count < line_count ) {
-                popup( _( "Spiked pits must be built over existing trenches!" ) );
-                return;
+    if( stop == overmap::invalid_tripoint ) {
+        return;
+    }
+    const recipe &making = recipe_id( miss_id.parameters ).obj();
+    bool change_x = start.x() != stop.x();
+    bool change_y = start.y() != stop.y();
+    if( change_x && change_y ) {
+        popup( _( "Construction line must be straight!" ) );
+        return;
+    }
+    if( miss_id.parameters == faction_wall_level_n_1_string ) {
+        std::vector<tripoint_abs_omt> tmp_line = line_to( stop, start );
+        // line_to doesn't include the origin point
+        tmp_line.emplace_back( stop );
```

Comment:
This fix snuck into this PR due to the whitespace changes. I can remove it to another PR if necessary.